test/e2e/all_test.go

```diff
@@ -0,0 +1,104 @@
+		t.Run("TestDynamicConfigManagerUnsupportedConfigOverride", TestDynamicConfigManagerUnsupportedConfigOverride)
+		t.Run("TestForwardedHeaderPolicyAppend", TestForwardedHeaderPolicyAppend)
+		t.Run("TestForwardedHeaderPolicyIfNone", TestForwardedHeaderPolicyIfNone)
+		t.Run("TestForwardedHeaderPolicyNever", TestForwardedHeaderPolicyNever)
+		t.Run("TestForwardedHeaderPolicyReplace", TestForwardedHeaderPolicyReplace)
+		t.Run("TestHAProxyTimeouts", TestHAProxyTimeouts)
+		t.Run("TestHAProxyTimeoutsRejection", TestHAProxyTimeoutsRejection)
+		t.Run("TestHTTPCookieCapture", TestHTTPCookieCapture)
+		t.Run("TestHTTPHeaderBufferSize", TestHTTPHeaderBufferSize)
+		t.Run("TestHTTPHeaderCapture", TestHTTPHeaderCapture)
+		t.Run("TestHeaderNameCaseAdjustment", TestHeaderNameCaseAdjustment)
+		t.Run("TestHealthCheckIntervalIngressController", TestHealthCheckIntervalIngressController)
+		t.Run("TestHostNetworkEndpointPublishingStrategy", TestHostNetworkEndpointPublishingStrategy)
+		t.Run("TestHostNetworkPortBinding", TestHostNetworkPortBinding)
+		t.Run("TestIngressControllerScale", TestIngressControllerScale)
+		t.Run("TestIngressControllerServiceNameCollision", TestIngressControllerServiceNameCollision)
+		t.Run("TestInternalLoadBalancer", TestInternalLoadBalancer)
+		t.Run("TestInternalLoadBalancerGlobalAccessGCP", TestInternalLoadBalancerGlobalAccessGCP)
+		t.Run("TestLoadBalancingAlgorithmUnsupportedConfigOverride", TestLoadBalancingAlgorithmUnsupportedConfigOverride)
+		t.Run("TestLocalWithFallbackOverrideForNodePortService", TestLocalWithFallbackOverrideForNodePortService)
+		t.Run("TestNetworkLoadBalancer", TestNetworkLoadBalancer)
+		t.Run("TestNodePortServiceEndpointPublishingStrategy", TestNodePortServiceEndpointPublishingStrategy)
+		t.Run("TestProxyProtocolAPI", TestProxyProtocolAPI)
+		t.Run("TestReloadIntervalUnsupportedConfigOverride", TestReloadIntervalUnsupportedConfigOverride)
+		t.Run("TestRouteAdmissionPolicy", TestRouteAdmissionPolicy)
+		t.Run("TestRouterCompressionParsing", TestRouterCompressionParsing)
+		t.Run("TestScopeChange", TestScopeChange)
+		t.Run("TestSyslogLogging", TestSyslogLogging)
+		t.Run("TestTLSSecurityProfile", TestTLSSecurityProfile)
+		t.Run("TestTunableMaxConnectionsInvalidValues", TestTunableMaxConnectionsInvalidValues)
+		t.Run("TestTunableMaxConnectionsValidValues", TestTunableMaxConnectionsValidValues)
+		t.Run("TestTunableRouterKubeletProbesForCustomIngressController", TestTunableRouterKubeletProbesForCustomIngressController)
+		t.Run("TestUniqueDomainRejection", TestUniqueDomainRejection)
+		t.Run("TestUniqueIdHeader", TestUniqueIdHeader)
+		t.Run("TestUserDefinedIngressController", TestUserDefinedIngressController)
+	})
+
+	t.Run("serial", func(t *testing.T) {
+		if os.Getenv("E2E_SKIP_SERIAL_TESTS") == "1" {
+			t.Skip()
+		}
+		t.Run("TestDefaultIngressControllerSteadyConditions", TestDefaultIngressControllerSteadyConditions)
+		t.Run("TestClientTLS", TestClientTLS)
```

Comment:
What happens if you don't have `t.Parallel` or `t.Series`? Is one assumed?

Specifically looking at `client_tls_test.go`, `configurable_route_test.go`, `dns_ingressdegrade_test.go`, `hard_stop_after_test.go`, etc..they don't have one. Should they?

Reply:
Without t.Parallel your test is not a candidate to be run in parallel. There is no t.Series; you have to opt-in to being parallel. You can also force all parallel tests to run in series if you invoke as `go test -parallel 1`



```
$ go help testflag
        -parallel n
            Allow parallel execution of test functions that call t.Parallel.
            The value of this flag is the maximum number of tests to run
            simultaneously; by default, it is set to the value of GOMAXPROCS.
            Note that -parallel only applies within a single test binary.
            The 'go test' command may run tests for different packages
            in parallel as well, according to the setting of the -p flag
            (see 'go help build').
```

Reply:
Whoops I don't know where I thought I saw `t.Series`, I think I just made that up in my head. Okay, good to know thanks!